libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SoqlSyncDownTarget.java

```diff
@@ -122,55 +128,68 @@ private JSONArray startFetch(SyncManager syncManager, long maxTimeStamp, String
         return records;
     }
 
+    protected JSONObject getResponseJson(RestResponse response) throws IOException {
+        JSONObject responseJson;
```

Comment:
In case of bad soql for instance, we get a json object with the error.
That would get lost. Now the error message from the server will end up in the exception.